lib/bundler/source/rubygems.rb

```diff
@@ -20,7 +20,7 @@ def initialize(options = {})
         @dependency_names = []
         @allow_remote = false
         @allow_cached = false
-        @caches = [Bundler.app_cache, *Bundler.rubygems.gem_cache]
+        @caches = [Bundler.app_cache, *Bundler.rubygems.gem_cache, Bundler.global_cache]
```

Comment:
we need to remove this change--the global cache should be for downloads only. It will be too slow to use the global cache for gem metadata, because getting the metadata out of .gem files means decompressing them and evaluating lots of ruby code.